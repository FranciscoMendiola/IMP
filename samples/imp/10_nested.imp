if x <= 10 then
  if y <= 5 then
    z := 1
  else
    z := 2
else
  while x <= 20 do
    x := x + 1
