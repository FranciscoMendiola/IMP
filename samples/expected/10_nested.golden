Token {tokenKind = KIf, lexeme = "if", position = 0}
Token {tokenKind = Id, lexeme = "x", position = 3}
Token {tokenKind = Leq, lexeme = "<=", position = 5}
Token {tokenKind = IntLit, lexeme = "10", position = 8}
Token {tokenKind = KThen, lexeme = "then", position = 11}
Token {tokenKind = KIf, lexeme = "if", position = 18}
Token {tokenKind = Id, lexeme = "y", position = 21}
Token {tokenKind = Leq, lexeme = "<=", position = 23}
Token {tokenKind = IntLit, lexeme = "5", position = 26}
Token {tokenKind = KThen, lexeme = "then", position = 28}
Token {tokenKind = Id, lexeme = "z", position = 37}
Token {tokenKind = Assign, lexeme = ":=", position = 39}
Token {tokenKind = IntLit, lexeme = "1", position = 42}
Token {tokenKind = KElse, lexeme = "else", position = 46}
Token {tokenKind = Id, lexeme = "z", position = 55}
Token {tokenKind = Assign, lexeme = ":=", position = 57}
Token {tokenKind = IntLit, lexeme = "2", position = 60}
Token {tokenKind = KElse, lexeme = "else", position = 62}
Token {tokenKind = KWhile, lexeme = "while", position = 69}
Token {tokenKind = Id, lexeme = "x", position = 75}
Token {tokenKind = Leq, lexeme = "<=", position = 77}
Token {tokenKind = IntLit, lexeme = "20", position = 80}
Token {tokenKind = KDo, lexeme = "do", position = 83}
Token {tokenKind = Id, lexeme = "x", position = 90}
Token {tokenKind = Assign, lexeme = ":=", position = 92}
Token {tokenKind = Id, lexeme = "x", position = 95}
Token {tokenKind = Plus, lexeme = "+", position = 97}
Token {tokenKind = IntLit, lexeme = "1", position = 99}
